src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -177,6 +177,11 @@ public static void daemonInitialization(Supplier<Config> config) throws Configur
 
         setConfig(config.get());
         applyAll();
+
+        createAllDirectories();
+        applyGuardrails(); // requires created directories
+        applyAfterDataDirectoriesAreCreated(); // requires setup Guardrails
```

Comment:
this is weird, why do we have a special method to apply something after data directories were created? I mean, we create directories before the main `applyGuardrails` so it seems to me that those two methods could merged inside Guardrails and do not introduce extra confusion in the API

Reply:
Yeah, odd; I will merge these together.